runtime/jvmti/jvmtiClass.c

```diff
@@ -237,39 +239,48 @@ jvmtiGetClassLoaderClasses(jvmtiEnv* env,
 
 		omrthread_monitor_enter(vm->classTableMutex);
 
-		memset(&stats, 0, sizeof(J9JVMTIClassStats));
-
 		stats.vm = vm;
 		stats.currentThread = currentThread;
 
 		/* Search for classes who have this class loader as the initiating loader (wind up count) */
-		clazz = vm->internalVMFunctions->hashClassTableStartDo(loader, &walkState);
+		clazz = vmFuncs->hashClassTableStartDo(loader, &hashWalkState);
 		while (clazz != NULL) {
 			countInitiatedClass(clazz, &stats);
-			clazz = vm->internalVMFunctions->hashClassTableNextDo(&walkState);
+			clazz = vmFuncs->hashClassTableNextDo(&hashWalkState);
 		}
 
+		/* The base type primitive array classes must also be added */
+		primitiveArray = &vm->booleanArrayClass;
+		do {
+			countInitiatedClass(*primitiveArray++, &stats);			
+		} while (primitiveArray <= &vm->longArrayClass);
+
 		stats.classRefs = j9mem_allocate_memory(stats.classCount * sizeof(jclass), J9MEM_CATEGORY_JVMTI_ALLOCATE);
 		if (stats.classRefs == NULL) {
 			rc = JVMTI_ERROR_OUT_OF_MEMORY;
 		} else {
-
 			/* Save count before it gets modified below... */
 			rv_class_count = (jint) stats.classCount;
 			rv_classes = stats.classRefs;
 
 			/* Record classes who have this class loader as the initiating loader (wind down count) */
-			clazz = vm->internalVMFunctions->hashClassTableStartDo(loader, &walkState);
+			clazz = vmFuncs->hashClassTableStartDo(loader, &hashWalkState);
 			while (clazz != NULL) {
 				copyInitiatedClass(clazz, &stats);
-				clazz = vm->internalVMFunctions->hashClassTableNextDo(&walkState);
+				clazz = vmFuncs->hashClassTableNextDo(&hashWalkState);
 			}
+
+			/* The base type primitive array classes must also be added */
+			primitiveArray = &vm->booleanArrayClass;
+			do {
+				copyInitiatedClass(*primitiveArray++, &stats);			
```

Comment:
I think we've been leaning towards retiring the use of `++` for non-primitive indexes and preferring an explicit ` += 1`.

Reply:
Done.